__tests__/index.test.ts

```diff
@@ -21,76 +20,79 @@ test('should export utils', () => {
   });
 });
 
-test('should be able to access properties on oas', () => {
-  expect(
-    new Oas({
-      info: { version: '1.0' },
-    }).api.info.version
-  ).toBe('1.0');
+test('should be able to access properties on the class instance', () => {
+  expect(new Oas(petstore as RMOAS.OASDocument).api.info.version).toBe('1.0.0');
 });
 
 describe('#getVersion()', () => {
-  it('should be able to identify a Swagger definition', () => {
-    expect(new Oas(swagger).getVersion()).toBe('2.0');
-  });
-
   it('should be able to identify an OpenAPI definition', () => {
-    expect(new Oas(petstore).getVersion()).toBe('3.0.0');
-    expect(new Oas(webhooks).getVersion()).toBe('3.1.0');
+    expect(new Oas(petstore as RMOAS.OASDocument).getVersion()).toBe('3.0.0');
+    expect(new Oas(webhooks as RMOAS.OASDocument).getVersion()).toBe('3.1.0');
```

Comment:
I loathe having to type these specs but there's some weird quirk with `openapi-types` and the petstore that we've got (that's a valid spec):

```
Argument of type '{ openapi: string; servers: { url: string; }[]; info: { description: string; version: string; title: string; termsOfService: string; contact: { email: string; }; license: { name: string; url: string; }; }; externalDocs: { ...; }; tags: ({ ...; } | { ...; })[]; paths: { ...; }; components: { ...; }; }' is not assignable to parameter of type 'OASDocument'.
  Type '{ openapi: string; servers: { url: string; }[]; info: { description: string; version: string; title: string; termsOfService: string; contact: { email: string; }; license: { name: string; url: string; }; }; externalDocs: { ...; }; tags: ({ ...; } | { ...; })[]; paths: { ...; }; components: { ...; }; }' is not assignable to type 'Omit<Omit<Document<{}>, "paths" | "components">, "paths" | "components" | "info" | "servers" | "webhooks" | "jsonSchemaDialect"> & { info: InfoObject; jsonSchemaDialect?: string; servers?: ServerObject[]; } & Pick<...> & Omit<...>'.
    Type '{ openapi: string; servers: { url: string; }[]; info: { description: string; version: string; title: string; termsOfService: string; contact: { email: string; }; license: { name: string; url: string; }; }; externalDocs: { ...; }; tags: ({ ...; } | { ...; })[]; paths: { ...; }; components: { ...; }; }' is not assignable to type 'Pick<PathsWebhooksComponents<{}>, "components">'.
      Types of property 'components' are incompatible.
        Type '{ schemas: { Order: { type: string; properties: { id: { type: string; format: string; }; petId: { type: string; format: string; }; quantity: { type: string; format: string; }; shipDate: { type: string; format: string; }; status: { ...; }; complete: { ...; }; }; xml: { ...; }; }; ... 4 more ...; ApiResponse: { ...; }...' is not assignable to type 'ComponentsObject'.
          Type '{ schemas: { Order: { type: string; properties: { id: { type: string; format: string; }; petId: { type: string; format: string; }; quantity: { type: string; format: string; }; shipDate: { type: string; format: string; }; status: { ...; }; complete: { ...; }; }; xml: { ...; }; }; ... 4 more ...; ApiResponse: { ...; }...' is not assignable to type '{ schemas?: Record<string, SchemaObject>; responses?: Record<string, ReferenceObject | ResponseObject>; parameters?: Record<...>; ... 6 more ...; pathItems?: Record<...>; }'.
            Types of property 'schemas' are incompatible.
              Type '{ Order: { type: string; properties: { id: { type: string; format: string; }; petId: { type: string; format: string; }; quantity: { type: string; format: string; }; shipDate: { type: string; format: string; }; status: { ...; }; complete: { ...; }; }; xml: { ...; }; }; ... 4 more ...; ApiResponse: { ...; }; }' is not assignable to type 'Record<string, SchemaObject>'.
                Property '"Order"' is incompatible with index signature.
                  Type '{ type: string; properties: { id: { type: string; format: string; }; petId: { type: string; format: string; }; quantity: { type: string; format: string; }; shipDate: { type: string; format: string; }; status: { type: string; description: string; enum: string[]; }; complete: { ...; }; }; xml: { ...; }; }' is not assignable to type 'SchemaObject'.
                    Type '{ type: string; properties: { id: { type: string; format: string; }; petId: { type: string; format: string; }; quantity: { type: string; format: string; }; shipDate: { type: string; format: string; }; status: { type: string; description: string; enum: string[]; }; complete: { ...; }; }; xml: { ...; }; }' is not assignable to type 'MixedSchemaObject'.
                      Types of property 'type' are incompatible.
                        Type 'string' is not assignable to type '("array" | NonArraySchemaObjectType)[]'.ts(2345)
```